src/components/ui/Button.tsx

```diff
@@ -19,19 +19,28 @@ export const ButtonWithState: React.FC<LoadableButtonProps> = ({
   onClick,
   children,
 }) => {
+  const [internalLoading, setInternalLoading] = React.useState(false);
   return (
     <button
       disabled={disabled || loading}
       className={`flex flex-row items-center justify-center gap-2 ${
-        (loading
+        (loading || internalLoading
           ? loadingClassName
           : disabled
           ? disabledClassName
           : className) ?? ""
       } ${loading || disabled ? "cursor-not-allowed" : ""}`}
-      onClick={onClick}
+      onClick={() => {
+        const result = onClick?.();
+        if (result instanceof Promise) {
+          setInternalLoading(true);
+          result.finally(() => setInternalLoading(false));
+        }
+      }}
```

Comment:
Do you mind explaining these lines a bit?

Reply:
If the function `onClick` returns a `Promise` instead of `void`, the button will start a loading animation until the function finishes. This is to avoid using an external loading state when we simply need to display loading for a single async function. See usage:

<img width="484" alt="Screenshot 2023-06-25 at 11 07 46 PM" src="https://github.com/ParachuteTeam/Parachute/assets/30245379/9f52bcc8-2885-4b32-a3c9-0e128bfa51b8">

(This example still uses a loading state to control the `Reset` button, but simply provides an async `onClick` function is sufficient for controlling the `Save` button)